libraries/botbuilder-runtime-integration-express/src/index.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as t from 'runtypes';
+import express from 'express';
+import { Configuration, getRuntimeServices } from 'botbuilder-runtime';
+import { IServices, ServiceCollection } from 'botbuilder-runtime-core';
+import { tests } from 'botbuilder-stdlib';
+
+/**
+ * Options for runtime Express adapter
+ */
+export type Options = {
+    /**
+     * Port that server should listen on
+     */
+    port: number;
+
+    /**
+     * Path that the server will listen to for [Activities](xref:botframework-schema.Activity)
+     */
+    messagingEndpointPath: string;
+};
+
+const defaultOptions: Options = {
+    messagingEndpointPath: '/api/messages',
+    port: 3978,
+};
+
+/**
+ * Start a bot using the runtime Express integration.
+ *
+ * @param applicationRoot application root directory
+ * @param settingsDirectory settings directory
+ * @param options options bag
+ */
+export async function start(
+    applicationRoot: string,
+    settingsDirectory: string,
+    options: Partial<Options> = {}
+): Promise<void> {
+    const [services, configuration] = await getRuntimeServices(applicationRoot, settingsDirectory);
+    const { port, messagingEndpointPath } = tests.isObject(options)
+        ? { ...defaultOptions, ...options }
+        : defaultOptions;
+
+    const app = await makeApp(messagingEndpointPath, services, configuration);
+    const { bot } = await services.mustMakeInstances('bot');
+
+    // The 'upgrade' event handler for processing WebSocket requests needs to be registered on the Node.js http.Server, not the Express.Application.
+    // In Express the underlying http.Server is made available after the app starts listening for requests.
+    const server = app.listen(port, () => {
+        console.log(`server listening on port ${port}`);
+    });
+
+    server.on('upgrade', async (req, socket, head) => {
```

Comment:
Why extract this piece from the `makeApp` function?

Reply:
Express doesn't return the `http.Server` until after the Application starts listening, so the upgrade event handler needs to be added in `start()`.

I added comments on [Lines 50-51](https://github.com/microsoft/botbuilder-js/pull/3354/files/257da017b05e4d2e5a05baab9f3e0dd119e7a247#diff-6008f79ef78560f59736ebbb604db3a42c1a292285a3708bbddde2e89fe933daR50-R51) to explain this, I can move part of the comments so it's right above the handler registration.

We could also turn `makeApp()` and `makeServer()` into private helpers to minimize surface area which makes `start()` appear to do the same thing to end users. This would be a better starting point for "v1" of these integration libraries since we can loosen the reins if customer requests come in.

Reply:
I'm sort of brainstorming here, but ideally all the logic for scaffolding a bot is hidden inside the `makeApp` or `makeServer` methods. That way, if a user decides to modify the app/server, it's purely additive and they don't have to copy and paste any code from the integration library.

Maybe `makeApp` should return `Promise<[app: express.Application, listen: (callback?: () => void) => http.Server>` and look like:
```typescript
function makeApp(...) ... {
  return [app, () => {
    const server = app.listen(options.port, callback ?? () => console.log(...));

    server.on('upgrade', () => ...);

    return server;
  }];
}
```

Then the caller does:

```typescript
const [app, listen] = await makeApp(...);
listen();
```